backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Product_BOM.java

```diff
@@ -109,4 +121,69 @@ public void validateBOMAttributes(final I_PP_Product_BOM productBom)
 		productPlanningDAO.retrieveProductPlanningForBomVersions(productBOMVersionsId)
 				.forEach(productPlanning -> productBOMService.verifyBOMAssignment(productPlanning, productBom));
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },
+			ifColumnsChanged = { I_PP_Product_BOM.COLUMNNAME_ValidFrom, I_PP_Product_BOM.COLUMNNAME_ValidTo })
+	public void preventBOMVersionsOverlapping(final I_PP_Product_BOM productBom)
+	{
+		if (productBom.getValidTo() != null && productBom.getValidTo().before(productBom.getValidFrom()))
+		{
+			throw new AdempiereException(MSG_VALID_TO_BEFORE_VALID_FROM);
+		}
+
+		final ProductBOMVersionsId bomVersionsId = ProductBOMVersionsId.ofRepoId(productBom.getPP_Product_BOMVersions_ID());
+		final BOMType bomType = BOMType.ofNullableCode(productBom.getBOMType());
+		final Optional<I_PP_Product_BOM> latestBOMVersions = bomDAO.getLatestBOMRecordByVersionAndType(bomVersionsId, ImmutableSet.of(bomType));
+
+		if (!latestBOMVersions.isPresent())
+		{
+			return;
+		}
+		else
+		{
+			final I_PP_Product_BOM latestBOMVersion = latestBOMVersions.get();
+
+			// Check if the BOM being updated is the same as the latest BOM version
+			if (productBom.getPP_Product_BOM_ID() != latestBOMVersion.getPP_Product_BOM_ID())
+			{
+				final Timestamp newValidFrom = productBom.getValidFrom();
+				final Timestamp newValidTo = productBom.getValidTo();
+				final Timestamp existingValidFrom = latestBOMVersion.getValidFrom();
+				final Timestamp existingValidTo = latestBOMVersion.getValidTo();
+
+				// Check for overlap
+				if (isOverlapping(newValidFrom, newValidTo, existingValidFrom, existingValidTo))
+				{
+					throw new AdempiereException(MSG_BOM_VERSIONS_OVERLAPPING,
+												 productBom.getName())
+							.markAsUserValidationError();
+				}
+			}
+		}
+	}
+
+	private boolean isOverlapping(final Timestamp newFrom, final Timestamp newTo, final Timestamp existingFrom, final Timestamp existingTo)
```

Comment:
pls
1. move it to TimeUtil and make it static
2. annotate params with `@Nullable` or `@NonNull`
3. provide unit tests including edge cases to understand if the intervals are [closed,open) or [closed,closed]. I would suggest to go with [closed,open) interval